File: python/structs/Queue.py
class Queue:
    def __init__(self):
        self.queue = []
        self.size = 0

    def enqueue(self, e):
        self.queue.append(e)
        self.size += 1

    def dequeue(self):
        temp = self.queue[0]
        for i in range(self.size - 1):
            self.queue[i] = self.queue[i + 1]
        self.size -= 1
        del self.queue[-1]

    def __str__(self):
        return self.queue.__str__()

    def __len__(self):
        return len(self.queue)


q = Queue()
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)

print(q)

q.dequeue()

print(q)

File: go/structs/Queue.go
package structs

import "fmt"

// Queue queue being created
type Queue struct {
	data []int
}

// Enqueue an item
func (q *Queue) Enqueue(item int) {
	q.data = append(q.data, item)
}

// IsEmpty check if the queue is empty
func (q Queue) IsEmpty() bool {
	return len(q.data) == 0
}

// Dequeue an item
func (q *Queue) Dequeue() int {
	if q.IsEmpty() {
		return -1
	}
	val := q.data[0]
	q.data = q.data[1:]
	return val
}

// Print the queue
func (q Queue) Print() {
	for i := 0; i < len(q.data); i++ {
		fmt.Printf("%d ", q.data[i])
	}
}

// Clear the queue
func (q *Queue) Clear() {
	q.data = q.data[:0]
}
